### ISSUE TITLE
False positive on checking a tuple element for None-ness

### ISSUE THREAD

**Issue by jolaf** [python/mypy#11479]
The following code:
```python3
from typing import Optional, Tuple

def f(t: Tuple[Optional[str]]) -> None:
    if t[0] and t[0].lower():
        print("OK")
    else:
        print("Empty")

f((None,))
f(('',))
f(('data',))
```
produces the following output:
```shell
$ python3 test.py
Empty
Empty
OK

$ mypy test.py
test.py: note: In function "f":
test.py:4:17: error: Item "None" of "Optional[str]" has no attribute "lower"  [union-attr]
        if t[0] and t[0].lower():
                    ^
$ mypy --version
mypy 0.910

$ python3 --version
Python 3.8.10

$ uname -a
Linux ubuntu 5.4.0-89-generic #100-Ubuntu SMP Fri Sep 24 14:50:10 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

$ cat /etc/issue
Ubuntu 20.04.3 LTS \n \l
```
`if t[0]` assumes that `t[0] is not None`, but mypy somehow misses it, which looks like an error.

if `if t[0]` is replaced with `if t[0] is not None`, mypy stops objecting.

If `Tuple[Optional[str]]` is replaced with just `Optional[str]`, and the rest of the code is updated accordingly, mypy stops objecting.

**Comment by hauntsaninja** [comment]
Thanks, I believe this is fixed on master
